### PR TITLE
Simple GRPC Client Server

### PR DESCRIPTION
**Pull request by brahmaroutu**
I have added two repos cosi-driver and cosi-provisioner. In this PR, I have a very simple GRPC call to getInfo working. 
cosi-driver> make container 
cosi-provisioner> make container
cosh-provisioner> kubectl create -f examples // create two pods one for GRPC server and the other for GRPC client. Makes a simple call to get driver identity.

### CODE REVIEW

**Comment by brahmaroutu** [comment]
@wlan0 @rrati Please review.

File: cosi-driver/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+* `--cosi-address`: This is the path to the COSI driver socket (defined above) inside the
+  pod that the `cosi-driver-provisioner` container will use to issue CSI
+  operations (e.g. `/cosi/cosi.sock`).
+
+### Required permissions
+
+The cosi-driver-sidecar does not interact with the Kubernetes API, so no RBAC
+rules are needed.
+
+It does, however, need to be able to mount hostPath volumes and have the file
```
Comment (wlan0) [inline cosi-driver/README.md]: Should we mention this? We are using ports instead of sockets

File: cosi-driver/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+## Community, discussion, contribution, and support
+
+Learn how to engage with the Kubernetes community on the [community page](http://kubernetes.io/community/).
+
+You can reach the maintainers of this project at:
+
+* Slack channels
+  * [#sig-storage](https://kubernetes.slack.com/messages/sig-storage)
+* [Mailing list](https://groups.google.com/forum/#!forum/kubernetes-sig-storage)
```
Comment (wlan0) [inline cosi-driver/README.md]: add wg-container-object-storage-interface

File: cosi-driver/SECURITY_CONTACTS [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# to for triaging and handling of incoming issues.
+#
+# The below names agree to abide by the
+# [Embargo Policy](https://github.com/kubernetes/sig-release/blob/master/security-release-process-documentation/security-release-process.md#embargo-policy)
+# and will be removed and replaced if they violate that agreement.
+#
+# DO NOT REPORT SECURITY VULNERABILITIES DIRECTLY TO THESE NAMES, FOLLOW THE
+# INSTRUCTIONS AT https://kubernetes.io/security/
+
+saad-ali
```
Comment (wlan0) [inline cosi-driver/SECURITY_CONTACTS]: our names

File: cosi-driver/cmd/cosi-driver/main.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```
Comment (wlan0) [inline cosi-driver/cmd/cosi-driver/main.go]: remove the cosh-driver binary pls.

File: cosi-driver/pkg/server/driverserver.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+type DriverServer struct {
+	Name, Version string
+}
+
+func (ds DriverServer) ProvisionerGetInfo(context.Context, *cosi.ProvisionerGetInfoRequest) (*cosi.ProvisionerGetInfoResponse, error) {
+	rsp := &cosi.ProvisionerGetInfoResponse{}
+	rsp.ProvisionerIdentity = fmt.Sprintf("%s-%s", ds.Name, ds.Version)
+	return rsp, nil //status.Error(codes.Unavailable, "Method not implemented")
```
Comment (wlan0) [inline cosi-driver/pkg/server/driverserver.go]: this looks good. Please remove the comment before we merge

**Comment by wlan0** [comment]
@brahmaroutu I think it'll be easier to review if the focus was only on the relevant changes in this PR.

Can you break this into two PRs, with the first one having the changes to the server and the second one the rest of the boiler plate around release process, docs and such.

**Comment by brahmaroutu** [comment]
@wlan0 Remove all the extra code. Had to keep most of release-tools for things to compile. Please review

**Comment by wlan0** [comment]
Thanks @brahmaroutu. Looks good.

The only change required is to remove the two binaries: cosi-driver/bin/cosi-driver, cosi-provisioner/bin/cosi-provisioner

File: cosi-driver/cmd/cosi-driver/main.go (hunk 4 above)
Comment (rrati) [reply cosi-driver/cmd/cosi-driver/main.go]: Need to update the copyright to 2020

File: cosi-driver/cmd/cosi-driver/main.go [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+	"github.com/container-object-storage-interface/spec/lib/go/cosi"
+)
+
+// Command line flags
+var (
+	cosiAddress = flag.String("cosi-address", "tcp://0.0.0.0:9000", "Path of the COSI driver socket that the provisioner  will connect to.")
+	showVersion = flag.Bool("version", false, "Show version.")
+	version     = "unknown"
+	// List of supported versions
+	supportedVersions = []string{"1.0.0"}
```
Comment (rrati) [inline cosi-driver/cmd/cosi-driver/main.go]: I feel like this should be 0.0.1 at this point. :)

File: cosi-driver/pkg/server/driverserver.go [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type DriverServer struct {
+	Name, Version string
+}
+
+func (ds DriverServer) ProvisionerGetInfo(context.Context, *cosi.ProvisionerGetInfoRequest) (*cosi.ProvisionerGetInfoResponse, error) {
```
Comment (rrati) [inline cosi-driver/pkg/server/driverserver.go]: These should be (ds *DriverServer) I believe.  Is there a reason for a value receiver vs over a pointer receiver?

File: cosi-driver/pkg/server/driverserver.go [hunk 8]
```diff
@@ -0,0 +1,47 @@
+package server
```
Comment (rrati) [inline cosi-driver/pkg/server/driverserver.go:17]: Add copyright

File: cosi-driver/release-tools/build.make [hunk 9]
```diff
@@ -0,0 +1,194 @@
+# Copyright 2019 The Kubernetes Authors.
```
Comment (rrati) [inline cosi-driver/release-tools/build.make]: Update the year

File: cosi-provisioner/LICENSE [hunk 10]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (rrati) [inline cosi-provisioner/LICENSE]: This should be top level file.  Entire repo is ASL2.0

**Review by rrati — APPROVED** [review]
Minor comments.  Update the copyright years for all files with copyright headers, and add headers to those files without headers

File: cosi-provisioner/LICENSE (hunk 10 above)
Comment (brahmaroutu) [reply cosi-provisioner/LICENSE]: removing it for now. we may need to have them later.

File: cosi-driver/release-tools/build.make (hunk 9 above)
Comment (brahmaroutu) [reply cosi-driver/release-tools/build.make]: done

File: cosi-driver/pkg/server/driverserver.go (hunk 8 above)
Comment (brahmaroutu) [reply cosi-driver/pkg/server/driverserver.go:17]: done

File: cosi-driver/pkg/server/driverserver.go (hunk 7 above)
Comment (brahmaroutu) [reply cosi-driver/pkg/server/driverserver.go]: Yep, moved to pointer receiver. No need to pass value object.

File: cosi-driver/cmd/cosi-driver/main.go (hunk 6 above)
Comment (brahmaroutu) [reply cosi-driver/cmd/cosi-driver/main.go]: done.

**Comment by brahmaroutu** [comment]
Thanks @rrati, Please review.

**Pull request merged**